packages/browser/src/core/queue/event-queue.ts

```diff
@@ -6,8 +6,14 @@ import { CoreEventQueue } from '@segment/analytics-core'
 import { isOffline } from '../connection'
 
 export class EventQueue extends CoreEventQueue<Context, AnyBrowserPlugin> {
-  constructor(priorityQueue?: PriorityQueue<Context>) {
-    super(priorityQueue ?? new PersistedPriorityQueue(4, 'event-queue'))
+  constructor(writeKey: string)
+  constructor(priorityQueue: PriorityQueue<Context>)
+  constructor(writeKeyOrQueue: string | PriorityQueue<Context>) {
+    super(
+      typeof writeKeyOrQueue === 'string'
+        ? new PersistedPriorityQueue(4, 'event-queue', writeKeyOrQueue)
+        : writeKeyOrQueue
+    )
```

Comment:
If we changed the signature, we could make this child class responsible for the writeKey.

something like:
writeKey: string, priorityQueue?: PriorityQueue